schema/main.go

```diff
@@ -19,6 +19,9 @@ var ErrLinkOrderNotSolvable = errors.New("link order is not solvable")
 // FileTag indicates the link should "use" the input file as source
 const FileTag = ":file"
 
+// SingleFileTag is a magic key indicating that a directory is actually a single file
+const SingleFileTag = ":single"
```

Comment:
@carsonfarmer the file vs. dir thing was getting pretty annoying. so, the api now _always_ expects a dir of files. If the schema demands a single file (no links), it will com in with a single link name `:single`, which is then parsed out.